library/core/src/main/java/com/google/android/exoplayer2/source/ConcatenatingMediaSource.java

```diff
@@ -297,6 +304,79 @@ public final synchronized void removeMediaSource(
     }
   }
 
+  /**
+   * Removes a range of {@link MediaSource}s from the playlist, by specifying an initial index
+   * (included) and a final index (excluded).
+   *
+   * <p>Note: when specified range is empty, no actual media source is removed and no exception
+   * is thrown.
+   *
+   * @param fromIndex The initial range index, pointing to the first media source that will be
+   *     removed. This index must be in the range of 0 &lt;= index &lt; {@link #getSize()}.
+   * @param toIndex The final range index, pointing to the first media source that will be left
+   *     untouched. The last media source to be removed is at index {@code toIndex} - 1.
+   *     This index must be in the range of 0 &lt;= index &lt; {@link #getSize()}.
+   *
+   * @throws IndexOutOfBoundsException when either index is out of playlist bounds, i.e. {@code
+   *      fromIndex} &lt; 0 or &gt;= {@link #getSize()}, {@code toIndex} &lt; 0 or &gt; {@link
+   *      #getSize()}
+   * @throws IndexOutOfBoundsException when range is malformed, i.e. {@code fromIndex} &gt;
```

Comment:
Can you combine them in one @throws doc? Maybe just as "When the range is malformed, i.e. {@code fromIndex} &lt; 0, {@code toIndex} &gt; {@link  #getSize()}, or {@code fromIndex} &gt; {@code toIndex}."

Reply:
Totally agree with grouping. Are you also suggesting to slim down the number of (broken) examples provided in JavaDoc?

Reply:
Yes. As we already require fromIndex>lastIndex, the other two examples are not needed.